micromanager_gui/main_window.py

```diff
@@ -594,8 +603,15 @@ def update_max_min(self, event=None):
 
     def snap(self):
         self.stop_live()
-        self._mmc.snapImage()
-        self.update_viewer(self._mmc.getImage())
+
+        # snap in a thread so we don't freeze UI when using process local mmc
+        @thread_worker(
+            connect={"finished": lambda: self.update_viewer(self._mmc.getImage())}
+        )
```

Comment:
what if instead of doing an actoin on "finished" we just connect to the new `imageSnapped` signal?

pro:

Allows synchronizing with a script (e.g. notebook) so when you snap there it also shows up in napari

cons:
maybe complicated when in an mda?
I suppose this draws us further away from other backends, although that is already probably a decent bit away.

Reply:
yeah, slightly wary of depending on the core-plus API at the moment.  Let's start with this.